src/main/java/org/generationcp/middleware/manager/UserProgramStateDataManagerImpl.java

```diff
@@ -1,28 +1,29 @@
 
 package org.generationcp.middleware.manager;
 
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.List;
-
 import org.generationcp.middleware.exceptions.MiddlewareQueryException;
 import org.generationcp.middleware.hibernate.HibernateSessionProvider;
 import org.generationcp.middleware.manager.api.UserProgramStateDataManager;
 import org.generationcp.middleware.pojos.UserProgramTreeState;
 import org.generationcp.middleware.util.Util;
-import org.hibernate.Criteria;
 import org.hibernate.HibernateException;
-import org.hibernate.criterion.Restrictions;
 import org.springframework.transaction.annotation.Transactional;
 
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
 /**
  * Created by cyrus on 12/16/14.
  */
 @Transactional
 public class UserProgramStateDataManagerImpl extends DataManager implements UserProgramStateDataManager {
 
+	private DaoFactory daoFactory;
+
 	public UserProgramStateDataManagerImpl(HibernateSessionProvider sessionProvider) {
 		super(sessionProvider);
+		this.daoFactory = new DaoFactory(sessionProvider);
 	}
```

Comment:
It seems we can also cut ties with DataManager here, the only link between the two is the logAndThrowException method. Something similar may apply for other classes extending from DataManager